### PR TITLE
Removed Equatable constraint from Storable

### PR DESCRIPTION
**Pull request by MP0w**
It's aa huge gain for usability and the loss is not that big I think
To be discussed

### CODE REVIEW

File: Tests/KakapoDBTests.swift [hunk 1]
```diff
@@ -271,31 +274,39 @@ class KakapoDBTests: QuickSpec {
...
-                let elementToDelete = UserFactory(id: 2, db: sut)
+            it("should not delete an object with same id but different type") {
+                let userFactory = sut.create(UserFactory).first!
+                let elementToDelete = CommentFactory(text: "", likes: 0, id: userFactory.id)
                 expect{ try sut.delete(elementToDelete) }.to(throwError(errorType: KakapoDBError.self))
-                expect(sut.findAll(UserFactory.self).count).to(equal(20))
+                expect(sut.findAll(UserFactory).count).to(equal(1))
+            }
+            
+            it("should delete an object with same id and different type but different properties") {
```
Comment (joanromano) [inline Tests/KakapoDBTests.swift]: shouldn't this be `should delete an object with *same id and type* but different properties` ?

File: Tests/KakapoDBTests.swift [hunk 2]
```diff
@@ -229,37 +229,40 @@ class KakapoDBTests: QuickSpec {
...
             it("should not update different kind of objects from different databases with same id") {
                 let anotherDB = KakapoDB()
-                sut.create(UserFactory.self, number: 20)
-                anotherDB.create(CommentFactory.self, number: 20)
-                expect{ try sut.update(anotherDB.find(CommentFactory.self, id: 5)!) }.to(throwError(errorType: KakapoDBError.self))
-                let updatedCommentInDb = sut.find(CommentFactory.self, id: 5)
+                sut.create(UserFactory.self, number: 2)
+                anotherDB.create(CommentFactory.self, number: 2)
+                expect{ try sut.update(anotherDB.find(CommentFactory.self, id: 0)!) }.to(throwError(errorType: KakapoDBError.self))
+                let updatedCommentInDb = sut.find(CommentFactory.self, id: 0)
```
Comment (joanromano) [inline Tests/KakapoDBTests.swift]: Why are we lowering all numbers in the tests?

**Comment by joanromano** [comment]
I think this might be a good idea but kind of a compromise for the future. Probably we should properly document what to expect/not to expect from deletion in db before merging this.

File: Tests/KakapoDBTests.swift (hunk 2 above)
Comment (MP0w) [inline Tests/KakapoDBTests.swift]: because it will be faster when all test creates 2/3 object instead of 20 and doesn't change anything

**Comment by MP0w** [comment]
@joanromano I added it to the `delete`documentation, is it fine?

**Comment by MP0w** [comment]
Btw there are only two way to delete the wrong thing: 
1. Creating an entity with same type and id: then you are supposed to understand what you are doing
2. Use another entity created by another DB (hard to happen)

**Comment by MP0w** [comment]
So I wouldn't be too concerned about it, if you are we might:
1. create another method for `Equatable` so can be used by equatable
2. refactor if to use the id directly instead of passing the entity itself.

File: Source/KakapoDB.swift [hunk 3]
```diff
@@ -157,15 +157,17 @@ public final class KakapoDB {
     /**
      Deletes the given Storable object
      
-     - parameter entity: The Storable object to be deleted
+     - parameter entity: The Storable object to be deleted. If an object with the same id and type can be found then will be deleted, even if the objects are not really equal.
```
Comment (joanromano) [inline Source/KakapoDB.swift]: I would rephrase this: `The Storable object to be deleted: the database will delete any object found with same id and type`

**Comment by joanromano** [comment]
👍
